src/fw-update/fw-update-device.cpp

```diff
@@ -112,6 +139,19 @@ namespace librealsense
         return false;
     }
 
+    float update_device::compute_progress(float progress, float start, float end, float threshold) const
+    {
+        return start + (ceil(progress * threshold) / threshold) * (end - start) / 100.f;
```

Comment:
devide by 0 protection?

Reply:
done (though this is a protected method, and I know which values are entered to this method)

Reply:
You know today :)